addons/mail/data/ir_cron_data.xml

```diff
@@ -77,5 +77,16 @@
             <field name="numbercall">-1</field>
             <field name="doall" eval="True"/>
         </record>
+        
+        <record id="ir_cron_clean_message_translation" model="ir.cron">
+            <field name="name">Message: Delete old translations</field>
+            <field name="interval_number">2</field>
+            <field name="interval_type">weeks</field>
+            <field name="numbercall">-1</field>
+            <field name="doall" eval="False"/>
+            <field name="model_id" ref="model_mail_message_translation"/>
+            <field name="code">model.search([]).unlink()</field>
```

Comment:
Rather than deleting everything every 2 weeks, it should run every day, to delete records that have more than 2 weeks.

Less costly operation more often is better to not lock the DB, and it's better functionally too to keep translations for 2 weeks from the moment they were generated.